config.go

```diff
@@ -27,14 +27,28 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
+// ReportingSamplingConfig offers way to configure reporting of sampling
+// events.
+//
+// With sampling there is always added ambiguity. A record is missing from the
```

Comment:
I think a better name may be `SamplingReportingConfig`

"This avoids the ambiguity of whether a record is missing because it was sampled, or whether it never took palce"